ibis/pyspark/compiler.py

```diff
@@ -1035,6 +1038,25 @@ def compile_join(t, expr, scope, timecontext, *, how):
     return left_df.join(right_df, pred_columns, how)
 
 
+def _canonicalize_interval(interval):
+    """ Convert interval to integer timestamp of second
+
+    When pyspark cast timestamp to integer type, it uses the number of seconds
+    since epoch. Therefore, we need cast ibis interval correspondingly.
+    """
+    if isinstance(interval, ir.IntervalScalar):
+        # value is in nanosecond
+        return int(execute(interval).value / 1e9)
```

Comment:
Do we have a translation rule for `IntervalScalar`? Perhaps we can add a rule for `IntervalScalar` and do

```
@compiles(IntervalScalar)
def compile_interval_scalar(interval):
    return execute(interval)

timedelta = t.translate(interval, ...) # This is a time
second_since_epoch = int(timedelta.value / 1e9)
```

Reply:
`IntervalScalar` is not of type `op` (operations), while all other @compiles are compiling `op`s. I found it a little strange to do so.

Reply:
`op` is `Literal`

Reply:
hmm i agree with @icexelloss here shouldn't we just be able to translate?

Reply:
or should this be a common method with other backends?

Reply:
I add this to compile(Literal)